src/interrupt.rs

```diff
@@ -2,9 +2,8 @@
 
 // NOTE: Adapted from cortex-m/src/interrupt.rs
 use crate::register::mstatus;
-pub use bare_metal::{CriticalSection, Mutex};
 
-/// Disables all interrupts
+/// Disables all interrupts in the current core.
```

Comment:
I wonder if we should specify what `core` means in RISC-V context here. e.g. is it a hard or an actual core with its own fetching unit.